scripts/python/airflow/dags/contralory_declaration_data_extraction.py

```diff
@@ -0,0 +1,194 @@
+def get_random_url():
+	return dag_services[randint(0, len(dag_services) - 1)]
+
+# arithmetic progression generator
+def ap(pos, first_item, second_item):
+	diff = second_item - first_item
+	index = pos
+
+	# generare index item
+	def generate():
+		nonlocal diff
+		nonlocal index
+		index += 1
+		return first_item + ((index - 1) - 1) * diff
+
+	return generate
+
+def list_navigator(cursor, ap, year, batch_size):
+	should_continue = True
+
+	while should_continue:
+		cursor.execute("""
+		SELECT drd.id as raw_id, ddf.file_name from staging.djbr_downloaded_files ddf
+		INNER JOIN staging.djbr_raw_data drd on ddf.raw_data_id = drd.id
+		WHERE drd.periodo >= %s
+		ORDER BY drd.id DESC -- begin with the latest declaration
+		OFFSET %s
+		LIMIT %s
+		""", [year, ap(), batch_size])
+
+		# fetchall to dictionary
+		desc = cursor.description
+		column_names = [col[0] for col in desc]
+		to_yield = [dict(zip(column_names, row)) for row in cursor.fetchall()]
+		yield to_yield
+
+		should_continue = len(to_yield) != 0
+
+def do_work(first_item, second_item, year, batch_size):
+	db_hook = PostgresHook(postgres_conn_id="postgres_default", schema="db")
+	db_conn = db_hook.get_conn()
+	db_cursor = db_conn.cursor()
+
+	page_size = ap(1, first_item, second_item)
+
+	insert_query = """
+	INSERT INTO analysis.ddjj_readable_data (
+		raw_data_id,
+		active,
+		passive,
+		net_worth,
+		parser
+	) VALUES (%s, %s, %s, %s, %s)
+	ON CONFLICT DO NOTHING;
+	"""
+
+	delete_query = """DELETE FROM staging.djbr_downloaded_files WHERE raw_data_id = %s"""
+
+	for rows in list_navigator(db_cursor, page_size, year, batch_size):
+		to_insert = []
+		to_delete = []
+
+		for row in rows:
+			t = Template(dumps(parser_request))
+			payload = t.substitute(resource=row['file_name'])
+			result = post(get_random_url(), data=payload)
+
+			if result.text is not None:
+				data = loads(result.text)
+				if 'data' in data.keys():
+					data = data['data']
+					if is_valid_data(data):
+						to_insert.append([row['raw_id'],data['resumen']['totalActivo'],
+						data['resumen']['totalPasivo'], data['resumen']['patrimonioNeto'],
+						"ms_djbr_parser"])
+					else:
+						to_delete.append([row['raw_id']])
+
+			if dag_timer > 0:
+				sleep(dag_timer)
+
+		if len(to_delete) > 0:
+			print(f"Sending {len(to_delete)}")
+			db_cursor.executemany(delete_query, to_delete)
+			db_conn.commit()
+
+		if len(to_insert) > 0:
+			print(f"Sending {len(to_insert)}")
+			db_cursor.executemany(insert_query, to_insert)
+			db_conn.commit()
+
+with dag:
+	launch = DummyOperator(task_id="start")
+	done = DummyOperator(task_id="done")
+
+	clean_db = PostgresOperator(task_id='clean_table',
+								sql="""
+									CREATE TABLE IF NOT EXISTS analysis.ddjj_readable_data (
+										id bigserial primary key,
+										raw_data_id bigint,
+										active numeric(20,2),
+										passive numeric(20,2),
+										net_worth numeric(20,2),
+										declaration_date timestamp,
+										parser text NULL,
```

Comment:
If this is the parsed data, rename it to 'parsed' and use the JSONB type.
```suggestion
										parsed JSONB,
```